### PR TITLE
Update french translation

### PR DESCRIPTION
**Pull request by bmfp**
Amost complete translation of Harbor portal
A few fields still have to be translated, but need code modifications to comply with french syntax (eg 3 days ago -> il y a 3 jours)

### CODE REVIEW

**Comment by jonasrosland** [comment]
This is great, thank you @bmfp for your PR!
As I'm not well-versed in the French language, I would like for someone who is take a look and verify if the language changes here are correct. So please hold, or if you know someone that would be a good fit for that, please tag them on this issue! We would be very greatful.

**Comment by bmfp** [comment]
Hi @jonasrosland 
Maybe @draeron and @julienvey could review

BR

File: src/portal/src/i18n/lang/fr-fr-lang.json [hunk 1]
```diff
@@ -45,59 +45,59 @@
...
-        "REPLICATE_FAILURE": "Started failed",
-        "STOP_SUCCESS": "Stop successfully",
-        "STOP_FAILURE": "Stop execution failed",
-        "TIME_OUT": "Gateway time-out"
+        "DELETED_SUCCESS": "Suppression effectuée",
+        "DELETED_FAILURE": "Suppression echouée",
+        "SWITCH_SUCCESS": "Bascule effectuée",
+        "SWITCH_FAILURE": "Bascule échouée",
+        "REPLICATE_SUCCESS": "Réplication effectuée",
+        "REPLICATE_FAILURE": "Réplication échoué",
```
Comment (julienvey) [inline src/portal/src/i18n/lang/fr-fr-lang.json:58]:
```suggestion
        "REPLICATE_FAILURE": "Réplication échouée",
```

File: src/portal/src/i18n/lang/fr-fr-lang.json [hunk 2]
```diff
@@ -45,59 +45,59 @@
...
-        "OVERRIDE": "Specify whether to override the resources at the destination if a resource with the same name exists.",
+        "NAME_FILTER": "Filtrer le nom de la ressource. Laissez vide ou utilisez '**' pour tout inclure. 'library/**' n'inclut que les ressources dans 'library'. Pour plus de patterns, référez-vous au guide d'utilisation.",
+        "TAG_FILTER": "Filtrer la partie tag/version des ressources. Laissez vide ou utilisez '**' pour tout inclure. '1.0*' n'inclut que les ressources qui commencent par '1.0'. Pour plus de patterns, référez-vous au guide d'utilisation.",
+        "LABEL_FILTER": "Filtrer les ressources en fonction des labels.",
+        "RESOURCE_FILTER": "Filtrer le type de ressources.",
+        "PUSH_BASED": "Pousser les ressources du Harbor local vers le registre distant.",
+        "PULL_BASED": "Tirer les ressources du registre distant vers le Harbor local.",
+        "DESTINATION_NAMESPACE": "Specifier le espace de nom de destination. Si vide, les ressources seront placées sous le même espace de nom que la source.",
```
Comment (julienvey) [inline src/portal/src/i18n/lang/fr-fr-lang.json:70]:
```suggestion
        "DESTINATION_NAMESPACE": "Specifier l'espace de nom de destination. Si vide, les ressources seront placées sous le même espace de nom que la source.",
```

File: src/portal/src/i18n/lang/fr-fr-lang.json [hunk 3]
```diff
@@ -45,59 +45,59 @@
...
-        "OIDC_ENDPOIT_FORMAT": "Endpoint must start with HTTPS://.",
-        "OIDC_NAME": "le nom du fournisseur de oidc.",
-        "OIDC_ENDPOINT": "l'url d'un serveur oidc plainte.",
-        "OIDC_SCOPE": "le champ envoyés au serveur au cours oidc l'authentification.il doit contenir 'openid', et 'offline_access'.si vous utilisez google, veuillez supprimer 'offline_access' dans ce domaine",
-        "OIDC_VERIFYCERT": "décocher cette case si votre oidc serveur est accueilli par auto - certificat signé.",
+        "NONEMPTY": "Ne peut pas être vide",
+        "REPO_TOOLTIP": "Les utilisateurs ne peuvent effectuer aucune opération sur les images dans ce mode.",
+        "ENDPOINT_FORMAT": "Le point de terminaison doit commencer par HTTP:// or HTTPS://.",
+        "OIDC_ENDPOIT_FORMAT": "Le point de terminaison doit commencer par HTTPS://.",
+        "OIDC_NAME": "Le nom du fournisseur de oidc.",
```
Comment (julienvey) [inline src/portal/src/i18n/lang/fr-fr-lang.json:94]:
```suggestion
        "OIDC_NAME": "Le nom du fournisseur oidc.",
```

File: src/portal/src/i18n/lang/fr-fr-lang.json [hunk 4]
```diff
@@ -45,59 +45,59 @@
...
+        "OIDC_ENDPOINT": "L'url d'un serveur de plainte oidc.",
+        "OIDC_SCOPE": "Le champ envoyé au serveur au cours de l'authentification oidc. Il doit contenir 'openid', et 'offline_access'. Si vous utilisez google, veuillez supprimer 'offline_access' dde ce champe",
```
Comment (julienvey) [inline src/portal/src/i18n/lang/fr-fr-lang.json:96]:
```suggestion
        "OIDC_SCOPE": "Le champ envoyé au serveur au cours de l'authentification oidc. Il doit contenir 'openid', et 'offline_access'. Si vous utilisez google, veuillez supprimer 'offline_access' de ce champ",
```

File: src/portal/src/i18n/lang/fr-fr-lang.json [hunk 5]
```diff
@@ -45,59 +45,59 @@
...
+        "OIDC_VERIFYCERT": "Décocher cette case si votre serveur oidc hébergé avec un certificat auto-signé.",
```
Comment (julienvey) [inline src/portal/src/i18n/lang/fr-fr-lang.json:97]:
```suggestion
        "OIDC_VERIFYCERT": "Décocher cette case si votre serveur oidc est hébergé avec un certificat auto-signé.",
```

File: src/portal/src/i18n/lang/fr-fr-lang.json [hunk 6]
```diff
@@ -180,26 +180,26 @@
...
         "DELETION_TITLE": "Confirmer la suppression utilisateur",
-        "DELETION_SUMMARY": "Voules-vous supprimer l'utilisateur {{param}}?",
+        "DELETION_SUMMARY": "Voules-vous supprimer l'utilisateur {{param}} ?",
         "DELETE_SUCCESS": "Utilisateur supprimé avec succès.",
         "ITEMS": "items",
-        "OF": "de",
-        "RESET_Ok": "Users password reset successfully",
-        "EXISTING_PASSWORD": "The new password can not be same with the old one"
+        "OF": "sur",
+        "RESET_Ok": "Réinitialisation du mot de passe de l'utilisateurs réussie",
```
Comment (julienvey) [inline src/portal/src/i18n/lang/fr-fr-lang.json:193]:
```suggestion
        "RESET_Ok": "Réinitialisation du mot de passe des utilisateurs réussie",
```

File: src/portal/src/i18n/lang/fr-fr-lang.json [hunk 7]
```diff
@@ -180,26 +180,26 @@
...
+        "EXISTING_PASSWORD": "Le nouveau mot de passe ne pzut pas être égal à l'ancien"
```
Comment (julienvey) [inline src/portal/src/i18n/lang/fr-fr-lang.json:194]:
```suggestion
        "EXISTING_PASSWORD": "Le nouveau mot de passe ne peut pas être égal à l'ancien"
```

File: src/portal/src/i18n/lang/fr-fr-lang.json [hunk 8]
```diff
@@ -225,17 +225,17 @@
...
-        "COUNT_QUOTA": "Count quota",
-        "STORAGE_QUOTA": "Storage quota",
-        "COUNT_QUOTA_TIP": "Please enter an integer between '1' & '100,000,000', '-1' for unlimited",
-        "STORAGE_QUOTA_TIP": "The upper limit of Storage Quota only takes integer values, capped at '1024TB'.  Enter '-1' for unlimited quota",
-        "QUOTA_UNLIMIT_TIP": "For unlimited quota, please enter '-1'."
+        "OF": "sur",
+        "COUNT_QUOTA": "Quota de nombre",
+        "STORAGE_QUOTA": "Quota de stockage",
+        "COUNT_QUOTA_TIP": "Entrez un entier entre '1' et '100000000', ou '-1' pour un quota illimité",
+        "STORAGE_QUOTA_TIP": "La limite haute du Quota de stockage n'accepte que des valrus entières, au maximum '1024TB'. Entrez '-1' pour un quota illimité",
```
Comment (julienvey) [inline src/portal/src/i18n/lang/fr-fr-lang.json:234]:
```suggestion
        "STORAGE_QUOTA_TIP": "La limite haute du Quota de stockage n'accepte que des valeurs entières, au maximum '1024TB'. Entrez '-1' pour un quota illimité",
```

File: src/portal/src/i18n/lang/fr-fr-lang.json [hunk 9]
```diff
@@ -557,46 +557,46 @@
...
         "JOB_PLACEHOLDER": "Nous n'avons trouvé aucun travail de réplication !",
-        "NO_ENDPOINT_INFO": "Please add an endpoint first",
-        "NO_PROJECT_INFO": "This project is not exist",
-        "SOURCE_RESOURCE_FILTER": "Source resource filter",
-        "SCHEDULED": "Scheduled",
-        "MANUAL": "Manual",
-        "EVENT_BASED": "Event Based",
-        "DAILY": "Daily",
-        "WEEKLY": "Weekly",
+        "NO_ENDPOINT_INFO": "Ajoutez un endpoint d'abord",
```
Comment (julienvey) [inline src/portal/src/i18n/lang/fr-fr-lang.json:565]:
```suggestion
        "NO_ENDPOINT_INFO": "Ajoutez d'abord un endpoint",
```

File: src/portal/src/i18n/lang/fr-fr-lang.json [hunk 10]
```diff
@@ -557,46 +557,46 @@
...
+        "TRIGGER_MODE": "Mode de déclenchement",
+        "SOURCE_PROJECT": "Projet source",
+        "REPLICATE": "Répliquer",
+        "DELETE_REMOTE_IMAGES": "Supprimer les ressources distantes lors de la suppression locale",
+        "DELETE_ENABLED": "Activer cette politique",
+        "NEW": "Nouvelle",
+        "NAME_TOOLTIP": "Le nom de la règle de réplication doit comporter au moins 2 caractères avec des minuscules, des chiffres et. _- et doit commencer par des caractères ou des chiffres.",
+        "DESTINATION_NAME_TOOLTIP": "Le nom de la destinationdoit comporter au moins 2 caractères avec des minuscules, des chiffres et. _- et doit commencer par des caractères ou des chiffres.",
+        "ACKNOWLEDGE": "Acquitter",
+        "RULE_DISABLED": "Cette règle a été désactivée car unlabel utilisé dans son filtre a été supprimé. \n Modifiez la règle et mettez à jour son filtre pour l'activer.",
```
Comment (julienvey) [inline src/portal/src/i18n/lang/fr-fr-lang.json:586]:
```suggestion
        "RULE_DISABLED": "Cette règle a été désactivée car un label utilisé dans son filtre a été supprimé. \n Modifiez la règle et mettez à jour son filtre pour l'activer.",
```

File: src/portal/src/i18n/lang/fr-fr-lang.json [hunk 11]
```diff
@@ -844,81 +844,81 @@
...
-            "OIDC_USERNAME": "d'utilisateur",
-            "GROUP_CLAIM_NAME": "Group Claim Name"
+            "OIDC_PROVIDER": "Fournisseur OIDC",
+            "OIDC_REDIREC_URL": "Vérifiez que la 'Redirect URI' du fournisseur OIDC est configurée",
+            "ENDPOINT": "Endpoint OIDC",
+            "CLIENT_ID": "ID client OIDC",
+            "CLIENTSECRET": "Client Secret OIDC",
+            "SCOPE": "Scope OIDC",
+            "OIDC_VERIFYCERT": "Vérifier le certificat",
+            "OIDC_SETNAME": "Configurer le nom d'utilisateur OICD",
```
Comment (julienvey) [inline src/portal/src/i18n/lang/fr-fr-lang.json:875]:
```suggestion
            "OIDC_SETNAME": "Configurer le nom d'utilisateur OIDC",
```

File: src/portal/src/i18n/lang/fr-fr-lang.json [hunk 12]
```diff
@@ -844,81 +844,81 @@
...
+            "OIDC_SETNAMECONTENT": "Vous devez créer un identifiant Harbor la première fois lors de la vérification par une tierce partie (OICD). Il sera utilisé au sein de Harbor pour être associé aux projets, rôles, etc.",
```
Comment (julienvey) [inline src/portal/src/i18n/lang/fr-fr-lang.json:876]:
```suggestion
            "OIDC_SETNAMECONTENT": "Vous devez créer un identifiant Harbor la première fois lors de la vérification par une tierce partie (OIDC). Il sera utilisé au sein de Harbor pour être associé aux projets, rôles, etc.",
```

File: src/portal/src/i18n/lang/fr-fr-lang.json [hunk 13]
```diff
@@ -1132,257 +1132,256 @@
...
-        "RETENTION_RUNS": "Retention runs",
-        "RUN_NOW": "RUN NOW",
-        "WHAT_IF_RUN": "DRY RUN",
-        "ABORT": "ABORT",
+        "TAG_RETENTION": "Rétention de Tags",
+        "RETENTION_RULES": "Règles de Rétention",
+        "RULE_NAME_1": " les images des {{number}} derniers jours",
+        "RULE_NAME_2": " les {{number}} images les plus actives",
+        "RULE_NAME_3": " les {{number}} images les plus récemment poussées",
+        "RULE_NAME_4": " les {{number}} images les plus récemment tirées",
```
Comment (julienvey) [inline src/portal/src/i18n/lang/fr-fr-lang.json:1200]:
```suggestion
        "RULE_NAME_4": " les {{number}} images les plus récemment récupérées",
```

**Comment by xaleeks** [comment]
@bmfp @julienvey  This is fantastic, thanks so much for doing this.  Out of curiosity, how developed is the technical verbiage around cloud native stuff in French?  I guess what I'm asking is are these the prevailing technical terms or more of a literal translation.  Either way, very insightful!

**Comment by julienvey** [comment]
@xaleeks It depends on who you ask :)

Personnaly, I prefer to stay close to the native english words, for instance when we talk about `push` or `pull` for git. It doesn't make sense to me to translate this in french. Others prefer strong translation. For instance, the French AWS Documentation is very "literal"

**Comment by bmfp** [comment]
I don't like too deep translations, and didn't do it this way
Some words are idiomatic, and understood and shared by the intended audience, I wanted to keep them to avoid misunderstanding of some content and misleading users

**Comment by xaleeks** [comment]
@bmfp @julienvey  Got it, and I wholeheartedly agree, whatever best facilitates understanding the action on the UI for the French speaking audience seems to be the best idea.  In Mandarin, we struggle with this as well, ie there's no official / accepted term for 'artifact' or 'helm' so we use whatever seems the most illuminating.  cheers

**Comment by AllForNothing** [comment]
@bmfp Please  rebase your code to resolve conflicts

**Comment by jonasrosland** [comment]
Hi @bmfp ! Thank you for updating your PR, can you please make sure you've signed off your commits?
Instructions can be found here: https://github.com/goharbor/harbor/pull/11653/checks?check_run_id=899174988

**Comment by bmfp** [comment]
looks like things have gone weird
it may be better to close this PR and create a new one with the only changed file in it

**Comment by wy65701436** [comment]
please rebase the PR, and make sure that there are only language changes. @renmaosheng will help to find a language expert to have a review.

**Comment by bmfp** [comment]
> 
> 
> thanks for contributing, please rebase the PR, and make sure that there are only language changes. @renmaosheng will help to find a language expert to have a review.

Hi @wy65701436, I don't know how to do this properly, could you give me some guidance/advice ?

thx

**Comment by sluetze** [comment]
> > thanks for contributing, please rebase the PR, and make sure that there are only language changes. @renmaosheng will help to find a language expert to have a review.
> 
> Hi @wy65701436, I don't know how to do this properly, could you give me some guidance/advice ?
> 
> thx

https://github.com/goharbor/harbor/blob/master/CONTRIBUTING.md

```
git checkout my_feature
git fetch -a
git rebase goharbor/master
```

**Comment by jonasrosland** [comment]
Hi @bmfp , I know this is a long standing PR, and I'm wondering if you would be willing to work on new French translation for the next release of Harbor? We are planning to release 2.3 some time next week, so I'm wondering if you would like to create the translation based off of 2.3 after it's released? We would be happy to get someone to review your contribution in a better time frame. You can also reach out to me @jonasrosland on the CNCF Slack.

**Pull request closed without merging**